mm2src/coins/z_coin/z_balance_streaming.rs

```diff
@@ -0,0 +1,110 @@
+use crate::common::Future01CompatExt;
+use crate::hd_wallet::AsyncMutex;
+use crate::z_coin::ZCoin;
+use crate::{MarketCoinOps, MmCoin};
+
+use async_trait::async_trait;
+use common::executor::{AbortSettings, SpawnAbortable};
+use common::log::{error, info};
+use futures::channel::mpsc::{UnboundedReceiver, UnboundedSender};
+use futures::channel::oneshot;
+use futures::channel::oneshot::{Receiver, Sender};
+use futures_util::StreamExt;
+use mm2_core::mm_ctx::MmArc;
+use mm2_event_stream::behaviour::{EventBehaviour, EventInitStatus};
+use mm2_event_stream::{Event, EventStreamConfiguration};
+use std::sync::Arc;
+
+pub type ZBalanceEventSender = UnboundedSender<()>;
+pub type ZBalanceEventHandler = Arc<AsyncMutex<UnboundedReceiver<()>>>;
+
+#[async_trait]
+impl EventBehaviour for ZCoin {
+    const EVENT_NAME: &'static str = "COIN_BALANCE";
+    const ERROR_EVENT_NAME: &'static str = "COIN_BALANCE_ERROR";
```

Comment:
Note for the future development: We may need to maintain all the available event names from a single source without needing to type them manually for each implementation.